### ISSUE TITLE
model saving/loading does not restore weights correctly

### ISSUE THREAD

**Issue by skeydan** [mlverse/torch#370]
Hi,

for a more complex model (this one here: https://github.com/mlverse/torchbook/blob/master/scripts/unet.R) saving & loading does not restore parameters correctly - or I'm doing something wrong :-)

After model training, before restarting the session, I see (for the example output right at the end of that file, which exactly represents what I've been executing):

```
> nnf_binary_cross_entropy(inferred_mask, mask$to(device = "cuda"))
torch_tensor
0.0130566
[ CUDAFloatType{} ]
> calc_dice_loss(inferred_mask, mask$to(device = "cuda"))
torch_tensor
 0.1684
[ CUDAFloatType{1} ]
```

After session restart & just creating an untrained model, I see:

```
> nnf_binary_cross_entropy(inferred_mask, mask$to(device = "cuda"))
torch_tensor
0.665109
[ CUDAFloatType{} ]
> calc_dice_loss(inferred_mask, mask$to(device = "cuda"))
torch_tensor
 1.0000
[ CUDAFloatType{1} ]
```

After model loading, again as per the code, I see

```
> nnf_binary_cross_entropy(inferred_mask, mask$to(device = "cuda"))
torch_tensor
0.666525
[ CUDAFloatType{} ]
> calc_dice_loss(inferred_mask, mask$to(device = "cuda"))
torch_tensor
 0.9596
[ CUDAFloatType{1} ]
```

which shows that something  changed but not much ...

Would you have an idea what could be going wrong? Thanks!

**Comment by dfalbel** [comment]
is your model in eval mode in both cases? there are many random stuff that can go inside models.

**Comment by skeydan** [comment]
shoot, I had wanted to test if this makes a difference :-(
back soon...

**Comment by skeydan** [comment]
:-( guess I should have "RTFM" as they say instead of just assuming `torch_load` was stateful like its equivalent in keras...
Closing.
